pkg/controller/incompatible_operator_controller.go

```
@@ -0,0 +1,168 @@
+)
+
+const (
+	reasonIncompatibleOperatorsInstalled = "IncompatibleOperatorsInstalled"
+	maxOpenShiftVersionProperty          = "olm.maxOpenShiftVersion"
+	ownerKindKey                         = "olm.operatorframework.io/owner-kind"
+	ownerNameKey                         = "olm.operatorframework.io/owner-name"
+	packageNameKey                       = "olm.operatorframework.io/package-name"
+	bundleNameKey                        = "olm.operatorframework.io/bundle-name"
+	bundleVersionKey                     = "olm.operatorframework.io/bundle-version"
+)
+
+type incompatibleOperatorController struct {
+	name                   string
+	nextOCPMinorVersion    *semver.Version
+	kubeclient             kubernetes.Interface
+	clusterExtensionClient *clients.ClusterExtensionClient
+	operatorClient         *clients.OperatorClient
+	logger                 logr.Logger
+}
+
+func NewIncompatibleOperatorController(name string, nextOCPMinorVersion *semver.Version, kubeclient kubernetes.Interface, clusterExtensionClient *clients.ClusterExtensionClient, operatorClient *clients.OperatorClient, eventRecorder events.Recorder) factory.Controller {
+	c := &incompatibleOperatorController{
+		name:                   name,
+		nextOCPMinorVersion:    nextOCPMinorVersion,
+		kubeclient:             kubeclient,
+		clusterExtensionClient: clusterExtensionClient,
+		operatorClient:         operatorClient,
+		logger:                 klog.NewKlogr().WithName(name),
+	}
+
+	return factory.New().WithSync(c.sync).WithSyncDegradedOnError(operatorClient).WithInformers(operatorClient.Informer(), clusterExtensionClient.Informer().Informer()).ToController(name, eventRecorder)
+}
+
+func (c *incompatibleOperatorController) sync(ctx context.Context, _ factory.SyncContext) error {
+	c.logger.Info("sync started")
+	defer c.logger.Info("sync finished")
+
+	incompatibleOperators, err := c.getIncompatibleOperators()
+	if err != nil {
+		c.logger.V(4).Error(err, "error checking incompatible operators")
```

Seems like we should propagate this error to the Operator status? Probably by setting `InstalledOLMOperatorsUpgradeable=Unknown` in this case?